compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -969,33 +968,21 @@ public BLangNode transform(ErrorTypeDescriptorNode errorTypeDescriptorNode) {
                 if (parent.kind() != SyntaxKind.TYPE_DEFINITION) {
                     return deSugarTypeAsUserDefType(errorType);
                 }
-            } else {
```

Comment:
Can we remove the `detail != null` check as well?

Reply:
done